pika/channel.py

```diff
@@ -332,11 +324,12 @@ def _generate_consumer_tag(self):
 
         :returns: consumer tag
         :rtype: str
+
         """
         return 'ctag%i.%s' % (self.channel_number,
                               uuid.uuid4().hex)
 
-    def basic_get(self, callback=None, queue='', no_ack=False):
+    def basic_get(self, callback, queue='', no_ack=False):
```

Comment:
`queue` shouldn't have a default value either, since you always need to pass a queue name, right?

Reply:
RabbitMQ "remembers" and uses the last auto-named queue used for a channel if you pass the empty string for an operation.

Reply:
Relying on it seems like a bug waiting to happen :). I wonder if, from the client API design point of view, it wouldn't be cleaner to get rid of these defaults? I don't know the answer, just wondering.